robocorp-code/vscode-client/src/inspector.ts

```diff
@@ -9,17 +9,34 @@ import { OUTPUT_CHANNEL } from "./channel";
 import { ChildProcess } from "child_process";
 import { feedback } from "./rcc";
 
-let _openingInspector: boolean = false;
-let _startingRootWindowNotified: boolean = false;
+export enum InspectorType {
+    Browser = "browser",
+    Windows = "windows",
+    Image = "image",
+    WebRecorder = "web-recorder",
+}
+
+export type InspectorTypes = `${InspectorType}`;
+
+export const DEFAULT_INSPECTOR_VALUE = {
+    browser: false,
+    image: false,
+    windows: false,
+    "web-recorder": false,
+};
+
+let _openingInspector: { [K in InspectorTypes]: boolean } = DEFAULT_INSPECTOR_VALUE;
```

Comment:
I'm not sure about this change. It was this way to prevent users from opening multiple inspectors at the same time -- I'm not sure about the current version, but I remember that opening say a browser recorder and an image recorder at the same time, if changing one the other wouldn't really know about such changes and I believe the user could even loose one thing recorded in one of those (given that they all use the same `locators.json`).

So, unless we actually can guarantee that this isn't the case, I'm not sure we should allow for that per-inspector -- although if this is now a valid scenario that was tested for, then that'd be ok ;)

Reply:
The Inspector treats the `locators.json` as a database now. 
It keeps a lock on it while writing & reading. 
The Picker app doesn't need a more forceful approach when dealing with the file. 
The Web Recorder has a polling system that keeps the database up to date & the save ability doesn't overlap with the Picker.

The user has the ability (with these changes) to open the Picker & the Web Recorder at the same time - but just one instance of each.  It would be close to impossible to create a race condition when file handling the `locators.json`.

From all the testing, it doesn't seem to be a problem while the two instances are opened. 
One improvement could be to just open the apps in the same process. 

This solution superficially solved a problem with hanging processes. While closing the Inspector the process hanged in the background, and as such you couldn't open another Inspector app because the process was considered in a running state. Fixed the issue afterwards, but the solution helped.

If still have concerns over this, I can revert some of the changes & keep a single process for the entire Inspectors spectrum of apps.

Reply:
Seems good to me then ;)